examples/conversation_persistor.py

```diff
@@ -0,0 +1,212 @@
+        self._log_q.put_nowait(None)
+        await self._main_task
+
+    def main_task(self) -> None:
+        # Listens for emitted MultimodalAgent events
+        @self._model.on("user_started_speaking")
+        def _user_started_speaking():
+            event = EventLog(eventname="user_started_speaking")
+            self._log_q.put_nowait(event)
+
+        @self._model.on("user_stopped_speaking")
+        def _user_stopped_speaking():
+            event = EventLog(eventname="user_stopped_speaking")
+            self._log_q.put_nowait(event)
+
+        @self._model.on("agent_started_speaking")
+        def _agent_started_speaking():
+            event = EventLog(eventname="agent_started_speaking")
+            self._log_q.put_nowait(event)
+
+        @self._model.on("agent_stopped_speaking")
+        def _agent_stopped_speaking():
+            transcription = TranscriptionLog(
+                role="agent",
+                transcription=(self._model._playing_handle._tr_fwd.played_text)[1:],
+            )
+            self._log_q.put_nowait(transcription)
+
+            event = EventLog(eventname="agent_stopped_speaking")
+            self._log_q.put_nowait(event)
+
+        @self._model.on("user_speech_committed")
+        def _user_speech_committed(user_msg: ChatMessage):
+            transcription = TranscriptionLog(
+                role="user", transcription=user_msg.content
+            )
+            self._log_q.put_nowait(transcription)
+
+            event = EventLog(eventname="user_speech_committed")
+            self._log_q.put_nowait(event)
+
+        @self._model.on("agent_speech_committed")
+        def _agent_speech_committed():
+            event = EventLog(eventname="agent_speech_committed")
+            self._log_q.put_nowait(event)
+
+        @self._model.on("agent_speech_interrupted")
+        def _agent_speech_interrupted():
+            event = EventLog(eventname="agent_speech_interrupted")
+            self._log_q.put_nowait(event)
+
+        @self._model.on("function_calls_collected")
+        def _function_calls_collected():
+            event = EventLog(eventname="function_calls_collected")
+            self._log_q.put_nowait(event)
+
+        @self._model.on("function_calls_finished")
+        def _function_calls_finished():
+            event = EventLog(eventname="function_calls_finished")
+            self._log_q.put_nowait(event)
+
+load_dotenv()
+
+logger = logging.getLogger("my-worker")
+logger.setLevel(logging.INFO)
+
+
+async def entrypoint(ctx: JobContext):
+    agent = multimodal.MultimodalAgent(
+        model=openai.realtime.RealtimeModel(
+            voice="alloy",
+            temperature=0.8,
+            instructions="You are a helpful assistant.",
+            turn_detection=openai.realtime.ServerVadOptions(
+                threshold=0.6, prefix_padding_ms=200, silence_duration_ms=500
+            ),
+        ),
+    )
+
+    cp = ConversationPersistor(model=agent, log="log.txt")
```

Comment:
ruff CI is failing due to this line not being used.
I think what we could do is to rename main_task() to start()
then call start() here?

Reply:
@theomonnom I updated per your suggestions! Just curious, is it usually better practice to not start the object upon instantiation or is it situational? I figured users would call upon cp in their own code so it wouldn't be a problem.

Reply:
It's generally better to avoid creating a new `asyncio.Task` during object instantiation. Moving the task creation into the start method offers a more flexible API. This allows for scenarios where you might want to configure or initialize an object without immediately starting it. 

By making the code more explicit, users of the API can confidently assume that no actions such as writing to a file will occur until the `start` method is explicitly called. There are no hidden or implicit behaviors.

For e.g, in our [SupervisedProcess](https://github.com/livekit/agents/blob/b6542abe5f42f5a49bc845301b57e91f049b7cd8/livekit-agents/livekit/agents/ipc/supervised_proc.py#L94) class, the object can be instantiated and initialized, but the actual OS process is only created when the `start` method is called.

Reply:
I thought it would be more efficient to have it all done upon instantiating but I definitely see how it's more flexible and user-friendly this way. Thank you for the feedback!